### PR TITLE
validate attribute of subscription

### PR DESCRIPTION
**Pull request by Letiste**

### CODE REVIEW

File: app/models/subscription.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class Subscription < ApplicationRecord
+  validates :duration, presence: true, numericality: { only_integer: true, greater_than: 0 }
+  validate :cannot_change_after_cancelled, :cannot_change_duration, on: :update
```
Comment (Letiste) [inline app/models/subscription.rb:5]: Should we also validate the date of cancelled_at? (for instance when set, it should not be in the past)

File: app/models/subscription.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  private
```
Comment (Letiste) [inline app/models/subscription.rb:7]: Should we handle these validations in setters? Or let it be handle during rails validation

Comment (nymous) [reply app/models/subscription.rb:7]: Let's keep rails validation for now!

File: app/models/subscription.rb (hunk 1 above)
Comment (nymous) [reply app/models/subscription.rb:5]: Let's not ^^

File: test/models/subscription_test.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  test 'duration must be strictly positive' do
+    subscription = Subscription.new(duration: 0)
+    assert_not_predicate subscription, :valid?
+
+    subscription = Subscription.new(duration: -1)
+    assert_not_predicate subscription, :valid?
+  end
+
+  test "canceled_at can't be changed when not nil" do
```
Comment (nymous) [inline test/models/subscription_test.rb:37]:
```suggestion
  test "cancelled_at can't be changed when not nil" do
```

File: app/models/subscription.rb (hunk 1 above)
Comment (Letiste) [reply app/models/subscription.rb:5]: No specific benefits from that

**Review by nymous — APPROVED** [review]

**Review by nymous — APPROVED** [review]

**Pull request merged**